reporter/symbol_uploader.go

```diff
@@ -253,9 +269,17 @@ func (d *DatadogSymbolUploader) Run(ctx context.Context) {
 				case <-ctx.Done():
 					return
 				case uploadData := <-d.uploadQueue:
-					if !d.upload(ctx, uploadData) {
-						// Remove from cache to retry later
+					extractedUploadData := d.GetSymbolsFromDisk(uploadData)
+					if extractedUploadData == nil {
 						d.uploadCache.Remove(uploadData.fileID)
+						return
```

Comment:
Just caught this while doing a more thorough review, this should be a `break` else we'd return from the goroutine and stop the worker:

```suggestion
						break
```

For example, with break we're skipping a message https://go.dev/play/p/n0bFjS2dS4Z. Whereas if we return early, we're returning from the entire function https://go.dev/play/p/FaQH8shGNdd - which in our case is the goroutine

Reply:
Good catch, thanks!